### PR TITLE
NDCG=0.0 instead of nan for IDCG=0.0

### PR DESCRIPTION
**Pull request by lironT74**
Hi, 
I believe this update is necessary, since if IDCG is 0 then NDCG should be 0, and not nan.

### CODE REVIEW

**Comment by guidozuc** [comment]
thanks @lironT74. This is a quite particular case. 
Mathematically, to compute nDCG, the DCG value is divided by the IDCG value. Then, if IDCG=0, x/0 = nan, i.e. dividing a number by zero is undefined. 
In practice, if IDCG=0, it means there are no relevant documents for that query in the qrels. Thus, one should exclude the query from the dataset or the evaluation: whatever you do on that query, will be impossible to evaluate (as you have no relevant documents for that query).

**Comment by ishnid** [comment]
I can see both sides :-)

I would suggest that the project should aim to follow whatever behaviour the official trec_eval has in situations like these.

**Comment by lironT74** [comment]
@guidozuc I agree, but as @ishnid mentioned, trec_eval outputs 0.0 for those cases.

File: trectools/trec_eval.py [hunk 1]
```diff
@@ -509,6 +509,9 @@ def get_ndcg(self, depth=1000, per_query=False, trec_eval=True, removeUnjudged=F
         idcg_per_query = perfect_ranking[["query",label]].groupby("query").sum()
         ndcg_per_query = dcg_per_query / idcg_per_query
 
+        #NDCG is 0 if IDCG=0
+        ndcg_per_query.fillna(0.0)
```
Comment (joaopalotti) [inline trectools/trec_eval.py:513]: This should have been:
ndcg_per_query = ndcg_per_query.fillna(0.0)
or 
ndcg_per_query.fillna(0.0, inplace=True)

Could you please fix that?
Thanks

**Review by joaopalotti — CHANGES_REQUESTED** [review]
Hi all,

Thanks for your comments, @lironT74, @ishnid and @guidozuc.
This issue here is somewhat close to #24. For the input that we got from there, we have recently modified some of the metrics to behave closer to ``trec_eval`` for particular corner cases like the one that you mentioned.

Unfortunately, your PR does not fix the problem as a whole. For example, ``get_ndcg`` with ``per_query==True`` would return a dataframe with all known queries (even the ones that are not in the run) and doing something like ``returned_df.mean()`` would result in a value that is different than running ``get_ndcg`` with ``per_query==False``. I have taken this into account in my latest try to fix this issue (please see this commit [here](https://github.com/joaopalotti/trectools/commit/a5785d71fc0170f173150ecb44c024c9736149ae)). 
But while it fixes the problem for NDCG, we still need to propagate this fix to other metrics. 
Would you be up for this challenge, @lironT74?

File: trectools/trec_eval.py (hunk 1 above)
Comment (lironT74) [reply trectools/trec_eval.py]: @joaopalotti sure, my bad, [fixed it here].(https://github.com/joaopalotti/trectools/pull/25/commits/b6ddb8214ca81fa57de5514cf9ae00cc64ea98f3)

**Comment by lironT74** [comment]
@joaopalotti I see. It seems that the fix is changing other metrics which might suffer from this issue the (almost exact) same way [you did for NDCG](https://github.com/joaopalotti/trectools/commit/a5785d71fc0170f173150ecb44c024c9736149ae) + the [fillna per query fix](https://github.com/joaopalotti/trectools/pull/25/commits/b6ddb8214ca81fa57de5514cf9ae00cc64ea98f3), correct?

I can work on it in my spare time, but I am not sure which additional metrics need to be changed.

**Comment by joaopalotti** [comment]
Thanks @lironT74, your suggestion was taken and included in the last commit!

**Pull request closed without merging**